crates/hotshot/src/lib.rs

```diff
@@ -639,6 +639,31 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, V: Versions> SystemContext<T
     }
 }
 
+#[cfg(feature = "hotshot-testing")]
+impl<TYPES: NodeType, I: NodeImplementation<TYPES>, V: Versions> SystemContext<TYPES, I, V> {
```

Comment:
I think we can just add this to the existing trait impl and enable the method when testing is enabled instead of doing separate bespoke impl.

Reply:
I think this should actually just go in `testing/src/helpers.rs` (ungated), next to `build_system_handle` (which is what we use currently in the tests). I'd call it `build_inactive_handle` or something (I think "lean" is slightly misleading, because it's just a dummy handle that isn't running anything).

I made an issue a while ago to do something like this for the non-integration tests (all the task-specific tests in `tests_1`), so I think this is good to have in general

Reply:
Good One @ss-es, I wanted to acknowledge the suggestion to place the test-specific helper function in `testing/src/helpers.rs`. During my implementation, I encountered issues due to the private and public(crate) visibility of dependent fields in pub struct SystemContext<> and pub struct SystemContextHandle<>. To avoid modifying the existing visibility of these fields, I opted for an alternative approach.